builder/builder.go

```diff
@@ -236,14 +271,19 @@ func (b *Builder) runStep(ctx context.Context, step *graph.Step) error {
 			step.Build = replacePositionalContext(step.Build, ".")
 		}
 		step.UpdateBuildStepWithDefaults()
-		args = b.getDockerRunArgs(volName, workingDirectory, step, step.Envs, "", "docker build "+step.Build)
+
+		if step.UseBuildCacheForBuildStep() {
+			args = b.getDockerRunArgs(volName, workingDirectory, step, step.Envs, "", buildxImg+" build "+step.Build, step.ID)
+		} else {
+			args = b.getDockerRunArgs(volName, workingDirectory, step, step.Envs, "", dockerImg+" build "+step.Build, step.ID)
+		}
```

Comment:
add new testcase in builder_test to cover the case.

Reply:
@northtyphoon I have already covered the unit test for UseBuildCacheForBuildStep in `step_test.go`.

Reply:
ping